### PR TITLE
adding package cli command

### PR DESCRIPTION
**Pull request by debopamsengupta**
Adding the `oc package` CLI command as per #419 
Also added a `--compress` option to generate the `package.tar.gz` file

### CODE REVIEW

**Comment by debopamsengupta** [comment]
cc @matteofigus 😄

**Comment by matteofigus** [comment]
Thanks @debopamsengupta sorry for taking some time. Perhaps @mattiaerre can help a bit with the review? If not I'll have a look tomorrow :)

**Comment by debopamsengupta** [comment]
No problem, thanks @matteofigus :)

File: test/unit/cli-facade-package.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+describe('cli : facade : package', function(){
+
+  var logSpy = {},
+      Registry = require('../../src/cli/domain/registry'),
+      registry = new Registry(),
+      Local = require('../../src/cli/domain/local'),
+      local = new Local({ logger: { log: function(){} } }),
+      readStub = sinon.stub().yields(null, 'test'),
+      PackageFacade = injectr('../../src/cli/facade/package.js', { read: readStub }),
```
Comment (mattiaerre) [inline test/unit/cli-facade-package.js]: you don't really need `injectr` here; you can just require `package`

File: test/unit/cli-facade-package.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+              expect(messageWithSlashesOnPath).to.include('components/hello-world/_package');
+              expect(logSpy.log.args[1]).to.be.undefined;
+              done();
+            });
+          });
+        });
+      });
+    });
+  });
+});
```
Comment (debopamsengupta) [inline test/unit/cli-facade-package.js]: should have improved now :)

**Comment by debopamsengupta** [comment]
@mattiaerre thanks for the comments :) made some improvements on the tests

**Comment by mattiaerre** [comment]
hi @debopamsengupta thanks for updating this PR; now we have got also 100% code coverage on that component which is very good. So this LGTM 👍 
I think we can merge this PR.
over to you @matteofigus

**Comment by matteofigus** [comment]
Thanks for working on this @debopamsengupta, very appreciated!

**Pull request merged**